### DISCUSSION TITLE
social sign-in - How to use a generic provider having a self signed signature?

### DISCUSSION THREAD

**Discussion by ymaiga** [ory/kratos#3049] [Q&A] [answered]
I have a hosted Kratos, I want to log with a generic OIDC provider having a non trusted signature. 
Is there a configuration in Ory Kratos to set a list of trusted certificates OR to disable the cert verification.

The error message I have : "Unable to initialize OpenID Connect Provider: Get \"https://my-provider/.well-known/openid-configuration\": x509: certificate signed by unknown authority"

**Comment by ymaiga** [comment, marked as answer]
I fix this point putting my ca certificate inside the `/etc/ssl/certs/` of Kratos container.

**Reply by vinckr** [reply]
Thanks for sharing your fix 🙌

**Comment by gammaSpeck** [comment]
Hello, @ymaiga and @vinckr. Were you able to able to authenticate users in an OIDC flow on a generic provider using x509 certificates?

The Ory docs [here](https://www.ory.sh/docs/kratos/concepts/credentials#:~:text=Other%20credentials%20%2D%20support%20other%20credential%20types%20(X509%20Certificates%2C%20Biometrics%2C%20...)%20that%20will%20be%20added%20at%20a%20later%20stage.) mentions that this feature is not yet supported. 

![image](https://github.com/ory/kratos/assets/39631861/e58ebc0a-de5d-407d-8b62-33805ee89c2b)


I am trying to configure authentication on Microsoft Azure AD using Certificates (We have a strict use case where we are not allowed to create client_secrets). 

I was able to follow this [microsoft article](https://learn.microsoft.com/en-us/answers/questions/1388776/how-to-get-access-token-using-certificate-based-au?source=docs) to be able to get back an access token. But, how do I configure the self hosted Kratos OIDC configs to support my use case?

**Reply by vinckr** [reply]
Hello @gammaSpeck 
Ory Kratos supports OIDC authentication with Microsoft Azure AD, but it requires a client secret as part of the configuration - AFAIK X509 certs are not supported. The steps to configure Microsoft Azure AD as a social sign-in provider are detailed in the [Ory documentation](https://www.ory.sh/docs/kratos/social-signin/microsoft).
